mobile/salto/src-tauri/src/commands/get_prices.rs

```diff
@@ -0,0 +1,131 @@
+use std::{env, time::Duration};
+use tauri::{AppHandle, Emitter, async_runtime};
+
+use crate::AppState;
+use crate::PriceConfig;
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Token {
+    currency: String,
+    value: f64,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Price {
+    symbol: String,
+    price: Vec<Token>,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct PriceResponce {
+    prices: Vec<Price>,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct CurrenciesResponce {
+    currencies: Vec<String>,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    Reqwest(#[from] reqwest::Error),
+    #[error(transparent)]
+    Json(#[from] serde_json::Error),
+    #[error("mutex lock error: {0}")]
+    Lock(String),
+}
+
+impl serde::Serialize for Error {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(self.to_string().as_ref())
+    }
+}
+
+#[tauri::command]
+pub async fn update_get_prices_config(
+    state: tauri::State<'_, AppState>,
+    currencies: Vec<String>,
+    new_assets: Option<Vec<String>>,
+) -> Result<(), Error> {
+    let assets = match new_assets {
+        Some(_) => Some(new_assets.unwrap()),
+        None => {
+            let cfg = state.get_prices_config.write().await;
+            cfg.assets.clone()
+        }
+    };
+    let mut cfg = state.get_prices_config.write().await;
+    *cfg = PriceConfig { currencies, assets };
+    Ok(())
+}
+
+#[tauri::command]
+pub async fn get_prices(
```

Comment:
I don't think this should be a command.
This should be a tokio task, run in the background with an automatic retry policy.
It should be emitting events that are listened to by the frontend.
See: https://tauri.app/develop/calling-frontend/#event-system